pkg/types/connectivitygroup.go

```diff
@@ -0,0 +1,226 @@
+package types
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+//go:generate callbackgen -type ConnectivityGroup
+type ConnectivityGroup struct {
+	*Connectivity
+
+	connections []*Connectivity
+	mu          sync.Mutex
+
+	states   map[*Connectivity]ConnectivityState
+	sumState ConnectivityState
+
+	connectCallbacks    []func()
+	disconnectCallbacks []func()
+	authCallbacks       []func()
+}
+
+type ConnectivityState int
+
+const (
+	ConnectivityStateDisconnected ConnectivityState = -1
+	ConnectivityStateUnknown      ConnectivityState = 0
+	ConnectivityStateConnected    ConnectivityState = 1
+	ConnectivityStateAuthed       ConnectivityState = 2
+)
+
+func getConnState(con *Connectivity) ConnectivityState {
+	state := ConnectivityStateDisconnected
```

Comment:
Should the default value be Unknown and add a disconnected else case?

Reply:
i later figured out that we can start with disconnected state